src/pages/index.astro

```diff
@@ -46,4 +46,7 @@ const { seo, ...dataWithoutSeo } = data;
       </main>
     </div>
   </body>
+  <script>
+    import '../scripts/updateHash';
```

Comment:
Jedyne co bym podmienił to wyeksportował z `updateHash` funkcję i takiej nazwie i wywołał ją tutaj przekazując `data`. Będzie to spójne z założeniem, że nie importujemy `data` nigdzie poza `index.astro`.